src/components/SplitButton/SplitButton.tsx

```diff
@@ -3,68 +3,90 @@ import React from 'react';
-import { DropMenuDumb as DropMenu } from '../DropMenu/DropMenu';
+import {
+	IDropMenuState,
+	IDropMenuProps,
+	DropMenuDumb as DropMenu,
+} from '../DropMenu/DropMenu';
 import * as reducers from './SplitButton.reducers';
 
 const cx = lucidClassNames.bind('&-SplitButton');
 
-const { any, bool, func, node, oneOf, shape, string } = PropTypes;
-
-const SplitButton = createClass({
-	displayName: 'SplitButton',
-
-	statics: {
-		peek: {
-			description: `
-				SplitButtons allow you to combine a single main \`Button\` together
-				with a list of additional Buttons with actions which will be rendered
-				within a DropMenu.
-			`,
-			categories: ['controls', 'buttons'],
-			madeFrom: ['Button', 'DropMenu'],
-		},
-	},
-
-	components: {
-		Button: createClass({
-			displayName: 'SplitButton.Button',
-			statics: {
-				peek: {
-					description: `
-						One of many potential \`Button\`s to render in this
-						\`SplitButton\`. The first \`Button\` will be used as the Primary
-						button, while all others will be rendered within the \`DropMenu\`
-						below.
-					`,
-				},
-			},
-			propTypes: {
-				children: any`
-					The children to render within the \`Button\`.
-				`,
-				isDisabled: bool`
-					Disables selection of the \`Button\`.
-				`,
-				onClick: func`
-					Called when the user clicks the \`Button\`.  Signature:
-					\`({ event, props }) => {}\`
-				`,
-			},
-		}),
-	},
-
-	reducers,
-
-	propTypes: {
+const { bool, func, node, oneOf, shape, string } = PropTypes;
+
+/** SplitButton Button Child Component */
+interface ISplitButtonButtonChildProps extends StandardProps {
+	/** Disables selection of the \`Button\`. */
+	isDisabled?: boolean;
+}
+
+const ButtonChild = (_props: ISplitButtonButtonChildProps): null => null;
+
+ButtonChild.displayName = 'SplitButton.ButtonChild';
+ButtonChild.peek = {
+	description: `
+		One of many potential \`Button\`s to render in this
+		\`SplitButton\`. The first \`Button\` will be used as the Primary
+		button, while all others will be rendered within the \`DropMenu\`
+		below.
+	`,
+};
+ButtonChild.propTypes = {
+	children: node`
+		The children to render within the \`Button\`.
+	`,
+	isDisabled: bool`
+		Disables selection of the \`Button\`.
+	`,
+	onClick: func`
+		Called when the user clicks the \`Button\`.  Signature:
+		\`({ props, event }) => {}\`
+	`,
+};
+
+export interface ISplitButtonProps extends StandardProps {
+	/** Sets the direction the flyout menu will render relative to the SplitButton. */
+	direction?: 'up' | 'down';
+
+	/** Style variations of the SplitButton. */
+	kind?: 'primary' | 'danger';
+
+	/** Size variations of the SplitButton. */
+	size?: 'short' | 'small' | 'large';
+
+	/** Form element type variations of SplitButton. Passed through to DOM Button. */
+	type?: string;
+
+	/** *Child Element* - props pass through to the underlying DropMenu component */
+	DropMenu: IDropMenuProps;
+}
+
+class SplitButton extends React.Component<ISplitButtonProps> {
```

Comment:
I think this is another component we could have converted to be functional (no lifecycle methods).